pydarn/plotting/grid.py

```diff
@@ -0,0 +1,261 @@
+# Copyright (C) 2020 SuperDARN Canada, University of Saskatchewan
+# Author: Daniel Billett, Marina Schmidt
+#
+# Modifications:
+#
+# Disclaimer:
+# pyDARN is under the LGPL v3 license found in the root directory LICENSE.md
+# Everyone is permitted to copy and distribute verbatim copies of this license
+# document, but changing it is not allowed.
+#
+# This version of the GNU Lesser General Public License incorporates the terms
+# and conditions of version 3 of the GNU General Public License,
+# supplemented by the additional permissions listed below.
+
+
+"""
+Grid plots, mapped to AACGM coordinates in a polar format
+"""
+
+import datetime as dt
+import matplotlib.pyplot as plt
+import numpy as np
+import cmocean
+
+from matplotlib import ticker, cm, colors
+from typing import List
+
+# Third party libraries
+import aacgmv2
+
+from pydarn import PyDARNColormaps, Fan, plot_exceptions
+
+
+class Grid():
+    """
+        Grid plots for SuperDARN data
+        This class inherits from matplotlib to generate the figures
+        Methods
+        -------
+        plot_grid
+    """
+    def __str__(self):
+        return "This class is static class that provides"\
+                " the following methods: \n"\
+                "   - plot_grid()\n"
+
+    @classmethod
+    def plot_grid(cls, dmap_data: List[dict], record: int = 0,
+                  start_time: dt.datetime = None, time_delta: int = 1,
+                  ax=None, fov: bool = True, parameter: str = 'vel',
+                  lowlat: int = 50, cmap: str = None, zmin: int = None,
+                  zmax: int = None, colorbar: bool = True,
+                  colorbar_label: str = '', title: str = '',
+                  len_factor: float = 150.0, ref_vector: int = 300):
+        """
+        Plots a radar's gridded vectors from a GRID file
+
+        Parameters
+        -----------
+            dmap_data: List[dict]
+                Named list of dictionaries obtained from SDarn_read
+            record: int
+                record number to plot
+                default: 0
+            start_time: datetime.datetime
+                datetime object as the start time of the record to plot
+                if none then record will be used
+                default: none
```

Comment:
Should you be able to use
```python
pydarn.Grid.plot_grid(grid_data)
```
and it pick out the first record for you and plot that? if so it can't, I get an error with it trying to find dtime later on if start_time is not given (line153).

Reply:
fixed